### ISSUE TITLE
`TextInputType` should support `webSearch` which maps to iOS `UIKeyboardType.webSearch`

### ISSUE THREAD

**Issue by stonemaster** [flutter/flutter#157562]
### Use case

iOS supports since iOS 7.0 the keyboard type `webSearch` which is a system keyboard that can be used in text fields that both search, but also cope with URLs. From the official documentation:

```objc
UIKeyboardTypeWebSearch API_AVAILABLE(ios(7.0)), // A default keyboard type with URL-oriented addition (shows space . prominently).
```

![image](https://github.com/user-attachments/assets/7423302d-0d3f-41fc-802f-fe97bbf9e7f6)

On Android this type could be mapped to `url` as it does pretty much the same, and has both space and . keys.

On iOS the `url` keyboard is useless for the search use case as it won't show a space bar.

I tried to workaround with custom iOS code but this is very tedious and I didn't find a solution.

Also there is a third-party package [keyboard_actions](https://pub.dev/packages/keyboard_actions) which allows to add a top-bar "." button, but this unfortauntely isn't the same user experience. Browsers like Safari, Chrome and Duck Duck Go use the iOS native  `UIKeyboardTypeWebSearch` in the search bars as well.

### Proposal

1. `TextInputType` should get a new enumeration `webSearch`
2. On Android (or any other platform) this remapped to `url`
3. The platform dependent code in `flutter/engine` for iOS is adapted to map `webSearch` to `UIKeyboardTypeWebSearch`

If there is a chance this is accepted by the Flutter team, I am gladly willing to do the implementation and create PRs for flutter and the engine.

**Comment by jomabe** [comment]
Yes, that would be extremely important! Without this point, no browser can be developed with Flutter for iOS.

**Comment by jmagman** [comment]
> On Android this type could be mapped to `url` as it does pretty much the same, and has both space and . keys.

In Chrome on Android when I type into the browser bar I see a normal-ish looking keyboard that has a space bar, but also prominently has the `/` and `.` at the bottom.

@reidbaker can you confirm that a text field that could be used for a URL or a web search (like the Chrome browser "Search or type URL") would correspond to `TYPE_TEXT_VARIATION_URI`?

@justinmc do you see any issue with adding a new enum corresponding to this case?

**Comment by reidbaker** [comment]
@jmagman confirmed. Best documentation on the subject I found here. https://developer.android.com/develop/ui/views/touch-and-input/creating-input-method

**Comment by jmagman** [comment]
> If there is a chance this is accepted by the Flutter team, I am gladly willing to do the implementation and create PRs for flutter and the engine.

@stonemaster thanks for the proposal and the offer for PRs, would be happy to review them!  https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md#developing-for-flutter

**Comment by stonemaster** [comment]
> @stonemaster thanks for the proposal and the offer for PRs, would be happy to review them! https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md#developing-for-flutter

Thank you very much for the quick response on this! I am travelling this week but will get my hands down on the PRs upcoming week 😄

**Comment by stonemaster** [comment]
One question: As this is merged into stable (thanks a lot to everybody), should I just close this issue? Or do you guys have some magic process that does it some time?

**Comment by jmagman** [comment]
@stonemaster yes it should be closed, thank you so much!

For reference next time you can put `Closes https://github.com/flutter/flutter/issues/157562` or `Fixes #157562` in your PR description and it will like the issue to the PR so it's closed automatically when it merges. https://github.com/flutter/flutter/issues/157562
